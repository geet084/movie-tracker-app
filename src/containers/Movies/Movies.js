import React, { Component } from 'react';
import Movie from '../Movie/Movie';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class Movies extends Component {
  
  render() {
    const { favorites, movies, location } = this.props;
    if (favorites && location.pathname.includes('favorites')) {
      return (
        <div id='movie-container' >
          {
            favorites.length ? favorites.map(movie => <Movie {...movie} key={movie.id} />) :
            <h4 className='no-faves'>No favorites to show... why not favorite some movies?</h4>
          }
        </div>
      )
    } else {
      return (
        <div id='movie-container'>
        {
          location.pathname.includes('favorites') ?
          <h4 className='faves-login'>Please <Link className='login' to='/login'>LOG IN </Link>to view favorites</h4> :
          movies.map(movie => <Movie {...movie} location={location} key={movie.id} />)
        }
        </div>
      )
    }
  }
}

Movies.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object),
  favorites: PropTypes.arrayOf(PropTypes.object),
  location: PropTypes.object,
}

Movies.defaultProps = {
  movies: [{}],
  location: {},
}

export const mapStateToProps = (state) => ({
  movies: state.movies,
  favorites: state.user.favorites,
});

export default connect(mapStateToProps)(Movies);