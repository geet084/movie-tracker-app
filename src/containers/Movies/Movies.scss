#movie-container {
  grid-gap: 20px;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  margin: 0 auto 10px auto;
  padding-top: 70px;
  @include display-type(grid, space-between, center);
  @include container-sizing(auto, 90%);
}

.faves-login {
  color: $main-text-color;
  font-size: 22px;
  text-align: center;
}

.login {
  text-decoration: none;
  color: $faves-page-login-color;
  &:hover {
    color: $hvr-text-color;
  }
}

.no-faves {
  color: $main-text-color;
  font-size: 22px;
  text-align: center;
}