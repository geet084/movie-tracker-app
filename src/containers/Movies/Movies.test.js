import React from 'react'
import { shallow } from 'enzyme'
import {Movies, mapStateToProps} from './Movies'

describe('Movies', () => {
  let wrapper;


  describe('defaults', () => {
    it('should match the initial snapshot', () => {
      const mockFavorites = undefined;
      const mockLocation = { pathname: '/' };
      wrapper = shallow(
        <Movies
          favorites={mockFavorites}
          location={mockLocation} />
      )
      expect(wrapper).toMatchSnapshot()
    })
    
    it('should match a favorites snapshot', () => {
      const mockFavorites = [{
        movie_id: 424694,
        release_date: "2018-10-24",
        title: "Bohemian Rhapsody",
        user_id: 44,
        vote_average: 8.2,
      }];
      const mockLocation = { pathname: '/favorites' };

      wrapper = shallow(
        <Movies
          favorites={mockFavorites}
          location={mockLocation} />
      )
      // TODO: WOULD LIKE TO SEE BOTH OUTPUTS TESTED...
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should convert store state to props ', () => {
      const mockMovie = {title: 'movie', id: 1}
      const mockState = { user: {favorites: []}, movies: [mockMovie]};
      const expected = {favorites: [], movies: [mockMovie]}
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected)
    })
  })
})