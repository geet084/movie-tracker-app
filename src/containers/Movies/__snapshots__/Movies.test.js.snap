// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies defaults should match a favorites snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movies
    favorites={
      Array [
        Object {
          "movie_id": 424694,
          "release_date": "2018-10-24",
          "title": "Bohemian Rhapsody",
          "user_id": 44,
          "vote_average": 8.2,
        },
      ]
    }
    location={
      Object {
        "pathname": "/favorites",
      }
    }
    movies={
      Array [
        Object {},
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Movie)
          movie_id={424694}
          release_date="2018-10-24"
          title="Bohemian Rhapsody"
          user_id={44}
          vote_average={8.2}
        />,
      ],
      "id": "movie-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "movie_id": 424694,
          "release_date": "2018-10-24",
          "title": "Bohemian Rhapsody",
          "user_id": 44,
          "vote_average": 8.2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Movie)
            movie_id={424694}
            release_date="2018-10-24"
            title="Bohemian Rhapsody"
            user_id={44}
            vote_average={8.2}
          />,
        ],
        "id": "movie-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "movie_id": 424694,
            "release_date": "2018-10-24",
            "title": "Bohemian Rhapsody",
            "user_id": 44,
            "vote_average": 8.2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Movies defaults should match the initial snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movies
    location={
      Object {
        "pathname": "/",
      }
    }
    movies={
      Array [
        Object {},
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Movie)
          location={
            Object {
              "pathname": "/",
            }
          }
        />,
      ],
      "id": "movie-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "location": Object {
            "pathname": "/",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Movie)
            location={
              Object {
                "pathname": "/",
              }
            }
          />,
        ],
        "id": "movie-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "location": Object {
              "pathname": "/",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
